rabbitmq/rabbitmq.go

```diff
@@ -0,0 +1,276 @@
+				if err != nil {
+					log.Errorf("Failed to start queue runner for %s on %s: %v", q.cfg.Name, spec.cfg.Name, err)
+					continue
+				}
+				queueRunners = append(queueRunners, r)
+			}
+		}
+		for _, r := range queueRunners {
+			r := r
+			g.Go(func() error { return r.run(gctx) })
+		}
+		// Signal readiness once topology is set and consumers are started
+		m.readyOnce.Do(func() { close(m.readyCh) })
+		// Monitor connection in the same errgroup; exit on context cancel too
+		g.Go(func() error {
+			select {
+			case <-gctx.Done():
+				return gctx.Err()
+			case err := <-m.connection.MonitorConnection():
+				if err != nil {
+					return err
+				}
+				return context.Canceled
+			}
+		})
+
+		if err := g.Wait(); err != nil && !errors.Is(err, context.Canceled) {
+			log.Errorf("Manager cycle error: %v", err)
+		}
+
+		// Close all queue runners and connection before next reconnect attempt
+		for _, r := range queueRunners {
+			r.close()
+		}
+		_ = m.connection.Close()
+	}
+}
+
+func buildExchangeSpecs(opts config.RabbitMQ) []ExchangeSpec {
+	var exchanges []ExchangeSpec
+
+	for _, configExchange := range opts.Exchanges {
+		var queues []QueueSpec
+
+		if configExchange.Name == "" || configExchange.Kind == "" {
+			log.Warnf("Skipping invalid exchange config (missing name/kind)")
+			continue
+		}
+
+		for _, configQueue := range configExchange.Queues {
+			if configQueue.Name == "" {
+				log.Warnf("Skipping invalid queue config on exchange %s: missing name", configExchange.Name)
+				continue
+			}
+			var handler Handler
+
+			// Map queue names to appropriate handlers
+			switch configQueue.Name {
+			case "password-change-queue":
+				handler = NewPasswordChangeHandler()
+			case "user-settings-updates":
+				handler = NewUserSettingsUpdateHandler()
+			}
+
+			if handler == nil {
+				log.Warnf("Skipping queue without handler on exchange %s: %s", configExchange.Name, configQueue.Name)
+				continue
+			}
+
+			// Resolve DLX/DLQ with queue-level override first, else exchange default
+			resolvedDLX := configQueue.DLXName
+			if resolvedDLX == "" {
+				resolvedDLX = configExchange.DLXName
+			}
+			resolvedDLQ := configQueue.DLQName
+			if resolvedDLQ == "" {
+				resolvedDLQ = configExchange.DLQName
+			}
+
+			queues = append(queues, QueueSpec{
+				cfg:     &configQueue,
+				Handler: handler,
+				dlxName: resolvedDLX,
+				dlqName: resolvedDLQ,
+			})
+		}
+
+		exchanges = append(exchanges, ExchangeSpec{
+			cfg:    &configExchange,
+			Queues: queues,
+		})
+	}
+
+	return exchanges
+}
+
+func declareRabbitMQExchange(conn *amqp.Connection, spec ExchangeSpec) error {
+	ch, err := conn.Channel()
+	if err != nil {
+		return fmt.Errorf("failed to open channel for exchange %s: %w", spec.cfg.Name, err)
+	}
+	defer ch.Close()
+
+	if err := ch.ExchangeDeclare(spec.cfg.Name, spec.cfg.Kind, spec.cfg.Durable, false, false, false, nil); err != nil {
+		return fmt.Errorf("failed to declare exchange %s: %w", spec.cfg.Name, err)
+	}
```

Comment:
Same thing as with queues: should we declare exchanges here?

I've read that they could be declared via "policies" and would expect them to be declared ahead of time.